pkg/providers/vsphere/virtualmachine/snapshot.go

```diff
@@ -174,69 +167,63 @@ func GetSnapshotSize(vmCtx pkgctx.VirtualMachineContext, vmSnapshot *vimtypes.Ma
-
-			// Add the disk files for the child most delta disk which is the last item in the disk chain.
-			for _, disk := range snapshot.Disk {
-				if fcdDeviceKeySet.Has(disk.Key) {
-					// A VM snapshot creates a delta disk for all disks of a VM -- including
-					// PVCs that are backed by FCDs. Since the usage of the delta disks
-					// created on PVCs will already be reported by the VolumeSnapshot
-					// SPU, we need to skip those here to avoid double counting.
-					vmCtx.Logger.V(5).Info("skipping the disk file of the FCD", "diskKey", disk.Key)
-					continue
-				}
+		// Add the file key for the snapshot memory (.vmem) file if present.
+		if snapshot.MemoryKey != -1 {
+			vmCtx.Logger.V(5).Info("Adding memoryKey", "memoryKey", snapshot.MemoryKey)
+			total += fileKeyMap[snapshot.MemoryKey]
+		}
 
-				if len(disk.Chain) == 0 {
-					vmCtx.Logger.V(5).Info("skipping the disk since its chain is empty", "diskKey", disk.Key)
-					continue
-				}
+		vmCtx.Logger.V(5).Info("Adding the file key for snapshot (.vmsn) file", "dataKey", snapshot.DataKey)
+		total += fileKeyMap[snapshot.DataKey]
+
+		// Add the disk files for the child most delta disk which is the last item in the disk chain.
+		for _, disk := range snapshot.Disk {
+			if fcdDeviceKeySet.Has(disk.Key) {
+				// A VM snapshot creates a delta disk for all disks of a VM -- including PVCs that are
+				// backed by FCDs. Since the usage of the delta disks created on PVCs will already be
+				// reported by the VolumeSnapshot SPU, we skip those here to avoid double counting.
+				vmCtx.Logger.V(5).Info("Skipping the disk file of the FCD", "diskKey", disk.Key)
+				continue
+			}
+
+			if len(disk.Chain) == 0 {
+				vmCtx.Logger.V(5).Info("Skipping the disk since its chain is empty", "diskKey", disk.Key)
+				continue
+			}
 
-				// file keys for the child most delta disk
-				childMostFileKeys := disk.Chain[len(disk.Chain)-1].FileKey
-				if len(childMostFileKeys) > 0 {
-					vmCtx.Logger.V(5).Info("Adding the file key for the child most delta disk of the snapshot",
-						"fileKey", childMostFileKeys)
-					fileKeyList = append(fileKeyList, childMostFileKeys...)
+			// File keys for the child most delta disk
+			childMostFileKeys := disk.Chain[len(disk.Chain)-1].FileKey
+			if len(childMostFileKeys) > 0 {
+				vmCtx.Logger.V(5).Info("Adding file key for the child most delta disk of the snapshot",
+					"fileKey", childMostFileKeys)
+				for _, fileKey := range childMostFileKeys {
+					total += fileKeyMap[fileKey]
 				}
 			}
 		}
-	}
-
-	vmCtx.Logger.V(5).Info("final fileKeyList", "fileKeyList", fileKeyList)
 
-	fileKeyMap := make(map[int32]int64)
-	for _, file := range vmlayout.File {
-		fileKeyMap[file.Key] = file.Size
+		break
 	}
 
-	var total int64
-	for _, fileKey := range fileKeyList {
-		if fileSize, ok := fileKeyMap[fileKey]; ok {
-			total += fileSize
-		}
-		// Assume the size is 0 if it's not found in the fileKeyMap
-	}
+	// TODO: Error if snapshot not found?
```

Comment:
IC, return early if `vmlayout.Snapshot` is empty. Ok will add it later

Reply:
Sure.